include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -393,6 +393,13 @@ class __future_base
     {
 #if !ONEDPL_ALLOW_DEFERRED_WAITING
         __my_event.wait();
```

Comment:
I'd suggest to make changes here. Why do we need a `wait_and_throw()` for `future`? I think this is about implementation for `future`, not about interfaces. On users side it can be just `wait()`, but we will implement it calling `wait_and_throw()`.
Just to remind it's a suggestion.

Any other opinions regarding it?

Reply:
Only issue is see is a mismatch in error propagation behavior between `sycl::event::wait()` and `dpl::future<T>::wait()`. Since `dpl::future<T>` is implicitly convertible to `sycl::event` this might lead to confusion. However, I like keeping the interface clean, like you're suggesting. We can address those issues by updating the documentation. I've updated the patch accordingly.